internal/policy/verify_test.go

```diff
@@ -728,65 +728,220 @@ func TestVerifyRelativeForRef(t *testing.T) {
+		entryID := common.CreateTestRSLReferenceEntryCommit(t, repo, entry, gpgKeyBytes)
+		entry.ID = entryID
+
+		err = VerifyRelativeForRef(context.Background(), repo, policyEntry, attEntry, policyEntry, entry, refName)
+		assert.Nil(t, err)
+
+		err = VerifyRelativeForRef(context.Background(), repo, policyEntry, nil, entry, policyEntry, refName)
+		assert.ErrorIs(t, err, rsl.ErrRSLEntryNotFound)
+	})
+
+	t.Run("no recovery, with bad state change", func(t *testing.T) {
+		repo, _ := createTestRepository(t, createTestStateWithPolicy)
+		refName := "refs/heads/main"
+
+		if err := repo.Storer.SetReference(plumbing.NewHashReference(plumbing.ReferenceName(refName), plumbing.ZeroHash)); err != nil {
+			t.Fatal(err)
+		}
+
+		initialPolicyEntry, _, err := rsl.GetLatestReferenceEntryForRef(repo, PolicyRef)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		repo, _ = createTestRepository(t, createTestStateWithPolicy)
+
+		// Create a new bad state which could not be constructed from the last state,
+		// since the signer of the bad state did not have permission to perform these
+		// actions in the initial state
+
+		signer, err := signerverifier.NewSignerVerifierFromSecureSystemsLibFormat(targets1KeyBytes) //nolint:staticcheck
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		key, err := tuf.LoadKeyFromBytes(targets1PubKeyBytes)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		rootMetadata := InitializeRootMetadata(key)
+
+		rootEnv, err := dsse.CreateEnvelope(rootMetadata)
+		if err != nil {
+			t.Fatal(err)
+		}
+		rootEnv, err = dsse.SignEnvelope(context.Background(), rootEnv, signer)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		newBadState := &State{
+			RootPublicKeys: []*tuf.Key{key},
+			RootEnvelope:   rootEnv,
+		}
+
+		if err := newBadState.Commit(repo, "Create test state", false); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := Apply(context.Background(), repo, false); err != nil {
+			t.Fatal(err)
+		}
+
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		commitIDs := common.AddNTestCommitsToSpecifiedRef(t, repo, refName, 1, gpgKeyBytes)
+		entry := rsl.NewReferenceEntry(refName, commitIDs[0])
+		entryID := common.CreateTestRSLReferenceEntryCommit(t, repo, entry, gpgKeyBytes)
+		entry.ID = entryID
+
+		err = VerifyRelativeForRef(context.Background(), repo, initialPolicyEntry, nil, initialPolicyEntry, entry, refName)
+		assert.ErrorIs(t, err, ErrVerifierConditionsUnmet)
+	})
+	t.Run("no recovery, with key revocation", func(t *testing.T) {
+		repo, _ := createTestRepository(t, createTestStateWithPolicy)
+		refName := "refs/heads/main"
+
+		targetsKey, err := tuf.LoadKeyFromBytes(targets1KeyBytes)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if err := repo.Storer.SetReference(plumbing.NewHashReference(plumbing.ReferenceName(refName), plumbing.ZeroHash)); err != nil {
+			t.Fatal(err)
+		}
+
+		initialPolicyEntry, _, err := rsl.GetLatestReferenceEntryForRef(repo, PolicyRef)
+
+		repo, _ = createTestRepository(t, createTestStateWithPolicy)
+
```

Comment:
two patterns:

1. `if err := ...; err !=nil` then there may or may not be a newline above, depending on what it's coming after.
2. OTOH, `..., err := ...\n if err != nil`, where the err check is purely an err check, no extra newline above.

Reply:
```suggestion
```

Reply:
Fixed